osutil/user.go

```diff
@@ -189,9 +192,16 @@ func sudoersFile(name string) string {
 	return filepath.Join(sudoersDotD, "create-user-"+strings.Replace(name, ".", "%2E", -1))
 }
 
+var hasAddUserExecutable = func() bool {
+	return ExecutableExists("adduser")
+}
+
 // AddUser uses the Debian/Ubuntu/derivative 'adduser' command for creating
 // regular login users on Ubuntu Core. 'adduser' is not portable cross-distro
 // but is convenient for creating regular login users.
+// if 'adduser' is not available, 'useradd' is used instead.
+// This order is preferred as older version of 'useradd' does not support
+// "--badname" option.
```

Comment:
this bit of comment doesn't relevant anymore?

Reply:
yes, since removing `--badname` this is indeed irrelevant. I did not know we even removed that option, that makes it more robust.... and makes me wonder why we did not use `useradd` from day one...

Reply:
removed